Spigot-Server-Patches/0317-New-EnderDragon-Events.patch

```diff
@@ -0,0 +1,66 @@
+From 22b245d1adccf8782b2a6ae9004f3c860f15c669 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Sun, 1 Jul 2018 11:50:05 -0500
+Subject: [PATCH] New EnderDragon Events
+
+
+diff --git a/src/main/java/net/minecraft/server/DragonControllerLandedFlame.java b/src/main/java/net/minecraft/server/DragonControllerLandedFlame.java
+index deee5c4cc..01c5f6527 100644
+--- a/src/main/java/net/minecraft/server/DragonControllerLandedFlame.java
++++ b/src/main/java/net/minecraft/server/DragonControllerLandedFlame.java
+@@ -64,6 +64,13 @@ public class DragonControllerLandedFlame extends AbstractDragonControllerLanded
+             this.d.setParticle(EnumParticle.DRAGON_BREATH);
+             this.d.a(new MobEffect(MobEffects.HARM));
+             this.a.world.addEntity(this.d);
++            // Paper start
++            com.destroystokyo.paper.event.entity.EnderDragonFlameEvent event = new com.destroystokyo.paper.event.entity.EnderDragonFlameEvent((org.bukkit.entity.EnderDragon) this.a.bukkitEntity, (org.bukkit.entity.AreaEffectCloud) this.d.bukkitEntity);
++            if (!event.callEvent()) {
++                this.a.world.removeEntity(this.d);
```

Comment:
Instead of this why not just avoid adding the entity to the world (eg line 14) unless the event wasn't cancelled.

Reply:
I tried that at first, but the [bukkitEntity is returning null](http://i.imgur.com/2soRT1t.png) if its not added to the world.

Reply:
Calling `getBukkitEntity()` should fix that, that field is populated lazily